syntax = "proto3";
option java_package="org.osgeo.mapguide";
package OSGeo.MapGuide;

import "Common.proto";
import "ResourceService_Messages.proto";

// Enables you to manipulate repositories and resources
service ResourceService {
    // Gets the content of the specified resource
    rpc GetResourceContent (GetResourceContentRequest) returns (GetResourceContentResponse);
    // Enumerates the resources in the specified repository
    rpc EnumerateResources (EnumerateResourcesRequest) returns (EnumerateResourcesResponse);
    // Checks to see if the specified resource exists
    rpc ResourceExists (ResourceExistsRequest) returns (ResourceExistsResponse);
    // Adds a new resource to a resource repository, or updates an existing resource
    rpc SetResource (SetResourceRequest) returns (BasicResponse);
}